src/JWT/Algorithms/RSAlgorithmFactory.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace JWT.Algorithms
+{
+    public sealed class RSAlgorithmFactory : HMACSHAAlgorithmFactory
+    {
+        private readonly Func<X509Certificate2> _certFactory;
+
+        public RSAlgorithmFactory(Func<X509Certificate2> certFactory)
```

Comment:
+1 passing a factory as a function to create the certFactory. This might turn out to be a generic `RSAlgorithmFactory` and not only rely on `X509Certificate2` if any other `RSAlgorithms` should be implemented.